lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,332 @@
+
+
+def fetch(module, content, vm):
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "src" not in module.params["fetch"]:
+        module.fail_json(msg="fetch.src is mandatory")
+    if "dest" not in module.params["fetch"]:
+        module.fail_json(msg="fetch.dest is mandatory")
+
+    dest = module.params["fetch"]['dest']
+    src = module.params['fetch']['src']
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_manager = content.guestOperationsManager.fileManager
+
+    fileTransferInfo = file_manager.InitiateFileTransferFromGuest(vm=vm, auth=creds,
+                                                                  guestFilePath=src)
+    url = fileTransferInfo.url
+    resp, info = urls.fetch_url(module, url, method="GET")
+    with open(dest, "wb") as local_file:
+        local_file.write(resp.read())
+    return True
+
+
+def copy(module, content, vm):
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "src" not in module.params["copy"]:
+        module.fail_json(msg="copy.src is mandatory")
+    if "dest" not in module.params["copy"]:
+        module.fail_json(msg="copy.dest is mandatory")
+    if "overwrite" in module.params["copy"]:
+        overwrite = module.params["copy"]["overwrite"]
+    else:
+        overwrite = False
+
+    dest = module.params["copy"]['dest']
+    src = module.params['copy']['src']
+    b_src = to_bytes(src, errors='surrogate_or_strict')
+    if not os.path.exists(b_src):
+        module.fail_json(msg="Source %s not found" % (src))
+    if not os.access(b_src, os.R_OK):
+        module.fail_json(msg="Source %s not readable" % (src))
+    if os.path.isdir(b_src):
+        module.fail_json(msg="copy does not support copy of directory: %s" % (src))
+
+    data = None
+    with open(b_src, "r") as local_file:
+        data = local_file.read()
+    file_size = os.path.getsize(b_src)
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_attributes = vim.vm.guest.FileManager.FileAttributes()
+    file_manager = content.guestOperationsManager.fileManager
+    url = file_manager.InitiateFileTransferToGuest(vm=vm, auth=creds, guestFilePath=dest,
+                                                   fileAttributes=file_attributes, overwrite=overwrite,
+                                                   fileSize=file_size)
+
+    resp, info = urls.fetch_url(module, url, data=data, method="PUT")
+    status_code = info["status"]
+    if status_code != 200:
+        raise Exception('initiateFileTransferToGuest : problem during file transfer')
+    return True
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(datacenter=dict(type='str'),
+                              cluster=dict(type='str'),
+                              folder=dict(type='str', default='/vm'),
```

Comment:
Setting default value for folder has create a lot of confusion in vmware_guest_* module, so I would recommend you to drop default value '/vm'.